### DISCUSSION TITLE
How to cope with ifcopenshell without installed mathutils?

### DISCUSSION THREAD

**Discussion by maciekspisak** [IfcOpenShell/IfcOpenShell#4889] [Q&A] [answered]
Hi, I am new to ifcopenshell (hope I am writing in proper place), however determined to develop my skills. Is it possible to use all features of ifcopenshell with python version >3.8 (i.e. without installed mathutils)? I have problems for example using ShapeBuilder.

**Comment by aothms** [comment, marked as answer]
https://github.com/IfcOpenShell/IfcOpenShell/blob/3c3e4dd0473a08443014867bc3d246115bd455db/src/ifcopenshell-python/ifcopenshell/util/shape_builder.py#L31

shape_builder is indeed one of (the only?) file that still directly depends on mathutils, because it is originally used in a blenderbim context. We are keen on rewriting this to numpy instead, but due to limited resources haven't been able to make progress on this front. If you're familiar with numpy I think this is a fairly trivial task. Perhaps you're interested in this to kickstart your ifcopenshell journey? We're happy to assist where needed.

**Reply by maciekspisak** [reply]
Thank you for the answer.
To be honest, when I was writing about starting my journey with ifcopenshell, I meant being the user. That is because I am not a developer, rather the engineer connected with structural engineering. I think that my coding skills could be not sufficient to write for others. However I will consider doing this after becoming familiar with the library.
